File: code/bubble_sort.cpp
#include <iostream>
using namespace std;

void bubble_sort(int n , int arr[]){
    for (int i = 0; i < n; i++)
    { 
        for (int j = 0; j < n -i-1; j++)
        {
            if (arr[j] > arr[j+1])
            {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
            
        }
        
    }
    for (int i = 0; i < n; i++)
    {
        cout  << arr[i] << " ";
    }
}

int main(){
    int n;
    cout << "Total Number of Element : ";
    cin >> n;
    cout << " Enter the Element of Array : ";

    int* arr = new int[n];
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    bubble_sort(n, arr); 
}

File: IT/bubblesort.java
import java.util.*;

public class bubblesort {
    public static void main(String[] arr) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter Size of Array:");
        int n = sc.nextInt();

        int count = 0;

        int[] a = new int[n]; 

        System.out.println("Enter the Elements of Array:");
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }

        for(int i = 0; i < n - 1; i++){
            for(int j = 0; j < n - i -1; j++){
                if(a[j] > a[j+1]){

                   int temp = a[j];
                    a[j] = a[j + 1];
                    a[j + 1] = temp;
                }
                count++;
            }
        }
        System.out.println("Sorted Array:");
        for (int i = 0; i < n; i++) {
            System.out.print(a[i] + " ");
        }

        System.out.println("\nTotal Comparisons: " + count);     
    }
}
